### ISSUE TITLE
10$: warn on array equality check

### ISSUE THREAD

**Issue by cgsdev0** [quick-lint/quick-lint-js#773]
example code:
```javascript
let arr = [];
if (arr === []) {
    // this can never happen
}
```

saw someone on twitch make this mistake in their code, when what they really wanted was `arr.length === 0`

maybe quicklint should warn that the expression `arr === []` here is always false?

**Comment by strager** [comment]
This diagnostic is a good idea. `=== {}` is similar.

**Comment by strager** [comment]
Yes. The comparison will always fail.

**Comment by nebulahh** [comment]
can i still work on this and what folder or file should I open to make the changes

**Comment by strager** [comment]
> can i still work on this

Yes.

> what folder or file should I open to make the changes

In `test/test-parse-warning.cpp` there are tests for similar cases. You can copy-paste and tweak as needed.

In `src/quick-lint-js/fe/parse.cpp` there's a `parser::error_on_sketchy_condition` function. That might be a good place to write the implementation.

`src/quick-lint-js/fe/expression.h` contains definitions for expression classes.

**Comment by cgsdev0** [comment]
I think this now duplicates https://github.com/quick-lint/quick-lint-js/issues/821

**Comment by strager** [comment]
@cgsdev0 Oh, you're right!

**Comment by strager** [comment]
@nebulahh You haven't showed progress on this task, and @Rebraws wrote a PR fixing this. I'm going to close this duplicate task and leave @Rebraws assigned.

Sorry for the confusion. I hope you didn't put in too much work already. =X
